### ISSUE TITLE
Cache response is not called on UITableView

### ISSUE THREAD

**Issue by abdulKarim002** [LeonardoCardoso/SwiftLinkPreview#55]
<li> Xcode 8.2
<li> Swift 3
<li> ios 9x - 10x

Everytime I call swift link preview it always call and set the data, how can I get the cache response, I am loading the preview data on UITableview so what should I do to get response from cache data when cell is recycled
` 
slp.preview(

                LinkObj.path,
                onSuccess: { result in
                //print("\(result)")

                cell.linkTitle.text = result[.title] as? String
                cell.linkURL.text = result[.canonicalUrl] as? String
                cell.linkDescription.text = result[.description] as? String
                
                let imageURL = result[.image] as? String
                print(imageURL!)
                cell.linkImage.kf.setImage(with: URL(string: imageURL!), placeholder: UIImage(named:""), options: [.transition(ImageTransition.fade(1))], progressBlock: nil, completionHandler: nil)
                
                cell.linkBtnRef.layer.setValue(result[.url], forKey: "linkURL")
                cell.linkBtnRef.addTarget(self, action: #selector(self.linkClickedOnLinkCell), for: .touchUpInside)
                
        },
            onError: { error in
                print("\(error)")
        }
        )`

**Comment by LeonardoCardoso** [comment]
You can cache the response, @abdulKarim002. Take a look at the following files: [Cache.swift](https://github.com/LeonardoCardoso/SwiftLinkPreview/blob/master/Sources/Cache.swift), [SwiftLinkPreview.swift#L46](https://github.com/LeonardoCardoso/SwiftLinkPreview/blob/master/Sources/SwiftLinkPreview.swift#L46) and [SwiftLinkPreview/pull/36](https://github.com/LeonardoCardoso/SwiftLinkPreview/pull/36).

**Comment by abdulKarim002** [comment]
What is the best way to access the cache protocol according to you @LeonardoCardoso  ? as i am using it for uitableview the spl.preview is already a async call so when you get the response from cache..

Thank you the response..

**Comment by pkc456** [comment]
+1

**Comment by goktugaral** [comment]
+1

**Comment by LeonardoCardoso** [comment]
Guys,

The default cache is disabled.
SLP has a built-in memory cache, so create your object as the following:

```swift
let slp = SwiftLinkPreview(cache: InMemoryCache())
```

To get the cached response:

```swift
if let cached = self.slp.cache.slp_getCachedResponse(url: String) {
    // Do whatever with the cached response
} else {
	// Perform preview otherwise
	slp.preview(...)
}
```

If you want to create your own cache, just implement this protocol and use it on the object initializer.

```
public protocol Cache {
    
    func slp_getCachedResponse(url: String) -> SwiftLinkPreview.Response?
    
    func slp_setCachedResponse(url: String, response: SwiftLinkPreview.Response?)
}
```

I've updated the documentation.
Let me know how it goes.

**Comment by goktugaral** [comment]
Hi @LeonardoCardoso ; 

I used your example, first I didn't see the result but after I changed my way it worked. The reason why it didn't work is, I'm using your component for my UITableViewCell with dequeueReusableCell. So, every time when l scrolled my tableview; cells are reloading (init/deinit) along with my previewView. That is why it didn't store cache.
Thank you for your time, it works great 👍
